### DISCUSSION TITLE
Search formulated string value

### DISCUSSION THREAD

**Discussion by leonlazic** [Budibase/budibase#12826] [Help] [answered]
I have a table of patients set up where I input a first name and a last name into a field. In order to make relating to other databases easier and to display it nicely in a table database I have set up another column where I concatenate the two strings into one.
I cannot figure out hot to make a search box where I could search either first name, last name or both.
![Screenshot](https://github.com/Budibase/budibase/assets/26551422/f897cd45-4842-4fe9-a0b1-3a00c0e8d02e)
![Screenshot](https://github.com/Budibase/budibase/assets/26551422/72406c05-f5d8-4d43-a0c1-42c94b553305)

I have been following [this guide](https://docs.budibase.com/docs/adding-a-search-field) using the second approach with the form and text input connected to a dataprovider filter.

But I cannot set the input text field to my formula field.
![Screenshot](https://github.com/Budibase/budibase/assets/26551422/08113acf-1650-4476-bbe8-2fe16d3fc1e1)

It kinda makes sense since this isn't possible because there is no logic on creating an input text field for a computed database field.
What would be the workaround for my case. Also note I want to have one search box to search both first and last name. I don't want a separate search box for each. I know how to make that.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @leonlazic,

You should be able to do this filtering on a formula. However this formula column needs to be set to static. Please note that because its static it will only run when one of the fields have been changed causing the value to update. (I'm just pointing this out as it will likely look like its not working when you create it initially, as all your data is already preset.)

Here are some screenshots of how I've set this column up in the data area.
<img width="358" alt="Screenshot 2024-01-22 at 09 13 02" src="https://github.com/Budibase/budibase/assets/126772285/1fb83c1a-786b-4f37-86ab-c3b8782b4465">
<img width="1004" alt="Screenshot 2024-01-22 at 09 13 18" src="https://github.com/Budibase/budibase/assets/126772285/2cfd0312-b54a-4cbb-bcd2-a9d387ccb4ee">

Additionally heres the table block settings showing were i've selected this column as the filter.
<img width="1512" alt="Screenshot 2024-01-22 at 09 14 25" src="https://github.com/Budibase/budibase/assets/126772285/c4534c39-9a03-4d32-ac0b-daae44a6851b">

Heres a demo of this in action and working.
![20240122_091912](https://github.com/Budibase/budibase/assets/126772285/32f8b68b-f31a-4049-9100-2516308edde7)

Additionally you can just create a raw fullName column with an automation. I've written up an example of this in a different discussion. You can find more on this [here](https://github.com/Budibase/budibase/discussions/10644#discussioncomment-5976751).

Hopefully this helps.

**Reply by leonlazic** [reply]
Thank you very much for your help. It worked. The only thing that had me stumped for a bit is that it works perfectly when i add a Table block and choose the name field from the search option of that block.
But if I want to do it manually then the name didn't appear in the text field box in the "Fields" dropdown.

Turns out that if I write my own name for that text box and call that in the handlebar of the dataprovider filter it will works great. But I still needed the static formula to be able to filter against that. Thanks for all the help.🙏

**Reply by ConorWebb96** [reply]
The fact you can see the static formula in the bindings area seems like a bug to me. However, I haven't been able to replicate this. Whenever I ejected the table block to confirm this, everything seemed to display correctly in the bindings area. (In my case i could see the static formula field `fullName` as well as the form field `fullName` to compare against within filters)

What version of Budibase are you running. I know there was an update really recently related to bindings. So this may have been indirectly fixed due to this.

Thanks in advance for the information.

**Reply by leonlazic** [reply]
I am extremely new to budibase. Started this past week. And I have already updated yesterday for the first time. I like keeping things up to date before reporting anything to avoid reporting old bugs. 
When ejecting the block the text field (search box) is also correctly named on my side but if you go to the drop down it isn't there.
As for the data provider filter the static formula does appear as it should.

My current app version is  2.15.4.

**Reply by ConorWebb96** [reply]
Oh right, I understand what you are talking about now. Thanks for the clarification!

Due to this being a formula field, I'm pretty sure its ignored as the fields inside this select are only ones that can be saved to. On the other hand as these fields hold values they can be filtered and therefore will appear in the list of columns to filter against.

I hope this makes sense.

**Reply by leonlazic** [reply]
Makes total sense. It also makes sense that it works without tying the actual search field box to the column in database since the functionality comes from the dataprovider filter. Thanks for everything.
